tests/test_filter_software_dependency.py

```diff
@@ -133,3 +133,55 @@ def test_not_valid_gcc_is_in_ubuntu_2004_d1(self):
                 reason_msg.getvalue(),
                 f"device compiler GCC {gcc_version} is not available in Ubuntu 22.04",
             )
+
+        def test_valid_cmake_versions_for_clangcuda_d2(self):
+            for cmake_version in [3.20, 3.26, 3.30, 3.49]:
```

Comment:
Evil little bug. `3.20` is a floating point number. Therefore it becomes `3.2` and than `CMAKE: ppv((CMAKE, cmake_version))` is called with `CMAKE: ppv((CMAKE, 3.2))`.
Change the version numbers to strings. This avoids that `3.20` becomes `3.2`.
```suggestion
            for cmake_version in ["3.20", "3.26", "3.30", "3.49"]:
```

Reply:
A debug hint: You can run a specific test with `python -m unittest discover -s tests -k <test_name>`, for example `python -m unittest discover -s tests -k test_valid_cmake_versions_for_clangcuda_d2`.

If you run only one test, print debugging is possible. If you put a `print(row)` at the begging of `software_dependency_filter()`, you see what is the input of you filter. This was the way, I found the bug. You can also put `print(row)` on different places in the filter function, to check which if statements your input passes and which not.